cmd/compute-domain-kubelet-plugin/driver.go

```diff
@@ -216,27 +215,11 @@ func (d *driver) nodeUnprepareResource(ctx context.Context, claimRef kubeletplug
 	d.Lock()
 	defer d.Unlock()
 
-	// Fetching the resource claim should not be needed (and not be done) in the
-	// unprepare code path. Any state required during unprepare can be stored
-	// via checkpointing.
-	claim, err := d.client.ResourceV1beta1().ResourceClaims(claimRef.Namespace).Get(
-		ctx,
-		claimRef.Name,
-		metav1.GetOptions{})
-
-	if err != nil {
-		return isPermanentError(err), fmt.Errorf("failed to fetch ResourceClaim %s in namespace %s: %w", claimRef.Name, claimRef.Namespace, err)
-	}
-
-	if claim.Status.Allocation == nil {
-		return true, fmt.Errorf("no allocation set in ResourceClaim %s in namespace %s", claim.Name, claim.Namespace)
```

Comment:
Was it possible for Unprepare to be invoked for a claim with `claim.Status.Allocation == nil`?

As the patch currently stands, we only Unprepare actively when the claim has been checkpointed, and we only checkpoint claims in Prepare, i.e. when `Status.Allocation != nil` (which I suppose is a bit of a stricter check, and when the claim has not been checkpointed we don't treat it as an error but as a success via noop-unprepare. Does that make sense?

Reply:
update, after confirming with Kevin: removal of the `if claim.Status.Allocation == nil` check makes sense.

About the noop-unprepare I am rather certain that's what we want but I want to check again with Kevin too.